Example/AsyncReactorExample/Features/Repository/Search/RepositorySearchReactor.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  TestReactor.swift
+//  Rudi
+//
+//  Created by Dominik Arnhof on 14.04.23.
+//
+
+import Foundation
+import AsyncReactor
+import UIKit
+import Logging
+
+private let logger = Logger(label: "RepositorySearchReactor")
+
+class RepositorySearchReactor: AsyncReactor {
+    enum Action {
+        case onHidePrivateToggle
+        case enterQuery(String)
+        case load
+    }
+    
+    struct State {
+        var hidePrivate = false
+        var query = ""
+        var repositories: [Repository] = []
+        var isLoading = false
+    }
+    
+    @Published
+    private(set) var state: State
+    
+    @MainActor
+    init(state: State = State()) {
+        self.state = state
+        
+        lifecycleTask {
+            for await _ in await NotificationCenter.default.publisher(for: UIApplication.didBecomeActiveNotification).values {
+                await self.handleNotification()
+            }
+            
+            logger.debug("lifecycleTask cancelled")
+        }
+    }
+    
+    func action(_ action: Action) async {
+        switch action {
+        case .onHidePrivateToggle:
+            state.hidePrivate.toggle()
+        case .enterQuery(let query):
+            state.query = query
+            
+            try? await Task.sleep(for: .seconds(5))
+            
+            guard !Task.isCancelled else { return }
+            
+            logger.debug("search: \(query)")
+        case .load:
+            Task {
+                state.isLoading = true
+                
+                do {
+                    let currentQuery = state.query.isEmpty ? "iOS" : state.query
+                    let (data, _) = try await URLSession.shared.data(from: URL(string:"https://api.github.com/search/repositories?q=\(currentQuery)")!)
+                    let decodedResponse = try? JSONDecoder().decode(RepositoriesResponse.self, from: data)
```

Comment:
You should remove the questionmark after the try so that parsing errors are also thrown and get logged in the catch block